source/game_sa/Scripts/TheScripts.cpp

```diff
@@ -1052,7 +1052,9 @@ CRunningScript* CTheScripts::StartNewScript(uint8* startIP, uint16 index) {
 
     script->RemoveScriptFromList(&pIdleScripts);
     script->Init();
-    script->SetCurrentIp(startIP);
+    if (startIP) { // NOTSA/BUGFIX: `IP` is initialized to null already, so no need to set it again (As it causes an assert)
```

Comment:
```suggestion
    if (startIP) { // NOTSA: `IP` is initialized to null already, so no need to set it again (As it causes an assert)
```
not a bug fix per se.

Reply:
Could the if be added inside SetCurrentIp?

Reply:
well that would defeat the whole point. though i'm not sure if a null check will be effective for this problem (reading invalid labels from script?), they'll be dangling mostly rather than 0.

Reply:
the point here is not to trigger the `assert` in `SetCurrentIP`.
this `if` is totally fine, because the IP is already set to `nullptr` in `Init()`